### PR TITLE
Merge harness

### PR DESCRIPTION
**Pull request by zhentaocc**
## Description

merge changes from https://github.com/analytics-zoo/lm-evaluation-harness into BigDL LLM Benchmark

### 1. Why the change?

Enable users to reproduce our benchmark results aligned with leaderboard settings.

### 2. User API changes

N/A

### 3. Summary of the change 

We implement a new `bigdl-llm` model type as harness official repo instructed, which uses bigdl.llm for int4, fp4, nf4, nf3,...

### 4. How to test?
- [ ] Unit test (Will add this in another PR.)

### CODE REVIEW

**Comment by zhentaocc** [comment]
@hkvision @jason-dai please review.

File: python/llm/dev/benchmark/harness/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+git checkout e81d3cc
+pip install -e .
+git apply ../bigdl-llm.patch
+cd ..
+```
+
+## Run
+run `python llb.py`. `llb.py` combines some arguments in `main.py` to make the tests easier. The mapping of arguments is defined as a dict in [`llb.py`](llb.py).
+
+### CPU Usage
```
Comment (hkvision) [inline python/llm/dev/benchmark/harness/README.md:19]: Evaluation on CPU

File: python/llm/dev/benchmark/harness/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+```python
+python llb.py --model bigdl-llm --pretrained /path/to/model --precision nf3 int4 nf4 --device cpu --tasks hellaswag arc mmlu truthfulqa --output_dir results/output
+```
+### Intel GPU Usage
```
Comment (hkvision) [inline python/llm/dev/benchmark/harness/README.md:23]: Evaluation on Intel GPU

File: python/llm/dev/benchmark/harness/bigdl-llm.patch [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+ from . import dummy
++from . import bigdl_llm
+ 
+ MODEL_REGISTRY = {
+     "hf": gpt2.HFLM,
+@@ -15,6 +16,7 @@ MODEL_REGISTRY = {
+     "anthropic": anthropic_llms.AnthropicLM,
+     "textsynth": textsynth.TextSynthLM,
+     "dummy": dummy.DummyLM,
++    "bigdl-llm": bigdl_llm.BigDLGPULM
```
Comment (hkvision) [inline python/llm/dev/benchmark/harness/bigdl-llm.patch:17]: better change this name, BigDLGPULM -> BigDLLLM?

**Review by hkvision — APPROVED** [review]

**Pull request merged**